<launch>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find planning)/config/rviz_sim1.rviz">
  </node>

  <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen">
    <param name="num_worker_threads" value="16"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="planning" args="load planning/Nodelet manager" output="screen">
    <!-- topic remap -->
    <remap from="~odom" to="odom"/>
    <remap from="~heartbeat" to="heartbeat"/>
    <remap from="~trajectory" to="trajectory"/>
    <remap from="~replanState" to="replanState"/>
    <remap from="~triger" to="/triger"/>
    <remap from="~target" to="/target/odom"/>

    <!-- init swarm state -->
    <rosparam file="$(find planning)/config/Swarm_config_Small.yaml" command="load" />

    <!-- replan params -->
    <param name="replan" value="true"/>
    <param name="plan_duration" value="2.0"/>
    <param name="kDeltaT" value="1.2"/>
    <param name="close_dist" value="2.0"/>
    <param name="close_time" value="4.0"/>
    <param name="TimeNoSafe" value="3.0"/>
      
    <!-- front-end params -->
    <param name="rho" value="3.5"/>

    <!-- MINCO Params -->
    <!-- origin: 8 5 8 -->
    <param name="K" value="6"/>
    <param name="Piece_num" value="5"/>
    <param name="FixK" value="6"/>
      
    <!-- constraints params -->
    <param name="vmax" value="1.7"/>
    <param name="amax" value="0.08"/>
    <param name="vmin" value="1.3"/>
    <param name="amin" value="-0.02"/>
    <param name="dSwarmMin" value="2.0"/>
    <param name="omegamax" value="0.30"/>

    <param name="rhoT" value="100.0"/>
    <param name="rhoV" value="1000.0"/>
    <param name="rhoA" value="100000.0"/>
    <param name="rhoAtail" value="100000.0"/>
    <param name="rhoVtail" value="1000.0"/>
    <param name="rhoPswarm" value="100000.0"/>
    <param name="rhoC" value="10000.0"/>
      
    <!-- optimizer monitor -->
    <param name="monitorUse" value="false"/>
    <param name="pausems" value="30"/>

  </node>

  <node pkg="planning" type="sim_node" name="sim_node" output="screen">
    <!-- topic remap -->
    <remap from="~trajectory" to="planning/Swarm_trajs"/>
    <remap from="~drone_states" to="planning/drone_states"/>
    <remap from="~target_traj" to="planning/target_traj"/>
    <remap from="~triger" to="/triger"/>

    <!-- safe sphere space visualization -->
    <param name="Safe_R" value="1.0"/>
  </node>
</launch>